source/extensions/filters/http/dynamic_modules/abi_impl.cc

```diff
@@ -50,6 +52,26 @@ bool headerAsAttribute(HeadersMapOptConstRef map, const Envoy::Http::LowerCaseSt
   return size > 0;
 }
 
+bool getFirstSanEntry(
+    OptRef<const Network::Connection> connection,
+    std::function<absl::Span<const std::string>(const Ssl::ConnectionInfoConstSharedPtr)>
+        get_san_func,
+    envoy_dynamic_module_type_buffer_envoy_ptr* result_buffer_ptr,
+    size_t* result_buffer_length_ptr) {
+  if (!connection.has_value() || !connection->ssl()) {
+    return false;
+  }
+  auto ssl = connection->ssl();
+  auto sans = get_san_func(ssl);
```

Comment:
can you specify explicit types and avoid using auto here? ack there's many places already but i would like to increase the usage unless absolute ncessary